### ISSUE TITLE
Allow editing share throws Error updating the share

### ISSUE THREAD

**Issue by edvanleeuwen** [nextcloud/server#21267]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Use Share link on a file
2. Click on the three points
3. Select Allow editing

### Expected behaviour
Editing access should be added.
The Allow editing should be ticked after refresh.

### Actual behaviour
Error updating the share.
The Allow editing is not ticked after refresh.

### Server configuration

Operating system:
Centos 7

Web server:
Apache

Database:
MariaDB

PHP version:
7.3

Nextcloud version: (see Nextcloud admin page)
18.0.5.1

**Updated from an older Nextcloud/ownCloud or fresh install:**
Update from 18.0.4.

**Where did you install Nextcloud from:**

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
Enabled:
  - cloud_federation_api: 1.1.0
  - dav: 1.14.0
  - documentserver_community: 0.1.5
  - federatedfilesharing: 1.8.0
  - files: 1.13.1
  - files_sharing: 1.10.1
  - lookup_server_connector: 1.6.0
  - oauth2: 1.6.0
  - onlyoffice: 4.2.0
  - provisioning_api: 1.8.0
  - settings: 1.0.0
  - twofactor_backupcodes: 1.7.0
  - viewer: 1.2.0
  - workflowengine: 2.0.0
Disabled:
  - accessibility
  - activity
  - admin_audit
  - circles
  - comments
  - encryption
  - federation
  - files_antivirus
  - files_external
  - files_fulltextsearch
  - files_pdfviewer
  - files_rightclick
  - files_trashbin
  - files_versions
  - files_videoplayer
  - firstrunwizard
  - fulltextsearch
 - files_pdfviewer
  - files_rightclick
  - files_trashbin
  - files_versions
  - files_videoplayer
  - firstrunwizard
  - fulltextsearch
  - group_everyone
  - impersonate
  - logreader
  - nextcloud_announcements
  - notifications
  - ownbackup
  - password_policy
  - photos
  - privacy
  - ransomware_detection
  - ransomware_protection
  - recommendations
  - serverinfo
  - sharebymail
  - sociallogin
  - spreed
  - support
  - survey_client
  - systemtags
  - text
  - theming
  - updatenotification
  - user_ldap
  - user_sql
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "bestanden.t.covs.nl"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/bestanden.t.covs.nl",
        "dbtype": "mysql",
        "version": "18.0.5.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "ownc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Europe\/Amsterdam",
        "installed": true,
        "maintenance": false,
        "theme": "",
        "loglevel": 0,
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": 0
        },
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "updater.release.channel": "stable",
        "enable_previews": true,
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\Markdown",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\Illustrator",
            "OC\\Preview\\Movie",
            "OC\\Preview\\MSOffice2003",
            "OC\\Preview\\MSOffice2007",
            "OC\\Preview\\MSOfficeDoc",
            "OC\\Preview\\OpenDocument",
            "OC\\Preview\\PDF",
            "OC\\Preview\\Photoshop",
            "OC\\Preview\\Postscript",
            "OC\\Preview\\StarOffice",
            "OC\\Preview\\SVG",
            "OC\\Preview\\TIFF"
        ],
        "mail_smtpauthtype": "LOGIN",
        "mail_sendmailmode": "smtp",
        "mail_smtpauth": 1,
        "mail_smtpsecure": "tls",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "data-fingerprint": "4dcfc356839b9f6a955b59898e5b2e54",
        "mysql.utf8mb4": true,
        "app_install_overwrite": [
            "impersonate",
            "spreed",
            "nextant"
        ],
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
No.

**Are you using encryption:** yes/no
No.

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No.

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**
Edge Chromium

**Operating system:**
Windows 10

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>
No details available. Loglevel set to 0.

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by kesselb** [comment]
Thank you for reporting this issue  :+1: 

I will need your browser console log to investigate this issue.  Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by edvanleeuwen** [comment]
[bestanden.t.covs.nl-1591457961355.log](https://github.com/nextcloud/server/files/4740552/bestanden.t.covs.nl-1591457961355.log)
![Aantekening 2020-06-06 174141](https://user-images.githubusercontent.com/7131152/83948456-937a3880-a81d-11ea-8390-060005dcfde0.png)

**Comment by rawtaz** [comment]
@edvanleeuwen Check if you have the setting under Sharing -> Allow users to share via link -> Allow public uploads unchecked. If it is unchecked, check it, then reload the page with your share and try to allow edits again. Let us know how it goes.

**Comment by edvanleeuwen** [comment]
> @edvanleeuwen Check if you have the setting under Sharing -> Allow users to share via link -> Allow public uploads unchecked. If it is unchecked, check it, then reload the page with your share and try to allow edits again. Let us know how it goes.

Tested and verified.
Checked the setting, reloaded the page, everything works.

**Comment by rawtaz** [comment]
After using Nextcloud for just a few days, I've noticed several cases where in the WUI you get some generic error message like "Error updating the share" or just "An unexpected error occurred", and when you dig into the actual network traffic you see that a more specific error message is actually being sent to the browser. Those error messages should be used, they'd probably mitigate a lot of problems people have. In this specific case it was clearly said in the error message sent to the browser that something like "public uploads not allowed", which lead me to that checkbox which solved the problem.

**Comment by kesselb** [comment]
cc @nickvergessen I think you fixed something similar?

**Comment by rawtaz** [comment]
Dudes, it's terribly inefficient and dumb that there's a bot closing perfectly valid issues just because project maintainers haven't had time to address them. I totally understand and respect if you haven't had the time to address this, but that should not mean the issue is closed automatically. When this happens, it'll just be forgotten and bugs will be lingering instead of being fixed.

**Comment by nickvergessen** [comment]
Dudette, it's terribly inefficient and dumb to be rude like this.

The issue is the label "needs info". It basically means info is missing. The info was provided, but the label not removed. And issues with that label are removed after X days without an answer. `¯\_(ツ)_/¯`

**Comment by kesselb** [comment]
I think this is already fixed by https://github.com/nextcloud/server/pull/21748 :thinking:

**Comment by rawtaz** [comment]
@nickvergessen I wasn't being rude, I was pointing out facts. If issues that haven't been resolved are closed, then obviously this doesn't lead to the software being improved. Hence, it's not efficient or good to have a bot that is unable to determine real world state of things go around and close issues. There's no need for a bot like that, there are actual humans that can close issues when it makes sense to do so. It would be different if the bot was actually intelligent, but just going by labels is evidently not working.

**Comment by nickvergessen** [comment]
Well as you might see on the issue number we are far over the 20.000 mark. We can't keep up with that without such assistance. Especially a lot of bug reports without proper reproduction steps or logs are not helpful at all. Either the author comes back with info (like you did) and then normally the person that set the label removes it again, or (sadly mostly the case) people never come back and we have tons of issues we can't reproduce and therefor not solve. Then we also don't need an open ticket about it.
Because when it happens to someone else, they can just report it again or comment on the previous issue abd we reopen it, no problem at all.

**Comment by rawtaz** [comment]
Fair enough, and again I understand (or at least expect) that you guys and gals are more or less overloaded, that's wouldn't be too uncommon for a project of this size.

On a related note, just because I have you on the hook temporarily, and because you seem to know a bit about how the project is managed on GitHub, could you ask someone to please answer https://github.com/nextcloud/server/pull/21797#issuecomment-679370535 ? Asking because I haven't been able to find an answer anywhere, and I'm pretty sure I'm not the only one wondering.

**Comment by skjnldsv** [comment]
I will need a screenshot of your network requests.
Before you do anything, still on the development tools, there is a tab called `network`. Click on it and then the xhr filter. Do your action and screenshot the network log like this:
![capture d ecran_2018-09-27_21-49-46](https://user-images.githubusercontent.com/14975046/46170857-43548000-c29f-11e8-9403-db72713dd9c1.png)

**Comment by pierreozoux** [comment]
I just found the same bug.

@kesselb said it was fixed on [this commit](https://github.com/nextcloud/server/pull/21748/commits/d9f12977d349a1cfd1b2f662b6e8e2ad861c8bbb) and it looks like it is just on Nc 21 and 20, and as I run on 19, I think that's normal then :)

I got this on the network tab:
<img width="524" alt="Screenshot 2021-05-17 at 21 33 04" src="https://user-images.githubusercontent.com/1863453/118545933-91239c80-b757-11eb-8a0b-76e78b419b63.png">

For me the bug is only the UI, it is not telling to the end user what is happening.

If it is fixed on 20 and 21, and you don't plan to backport to 19, then I propose to close it, I'll track it internally, and if I update to 20 and the bug is still present I'll come back to reopen.
